### ISSUE TITLE
cmd/compile/internal/gc: TestDeps can easily test incorrect version of go binary

### ISSUE THREAD

**Issue by dmitshur** [golang/go#34041]
```
$ go test cmd/compile/internal/gc
--- FAIL: TestDeps (1.24s)
    dep_test.go:23: undesired dependency on "go/build"
    dep_test.go:23: undesired dependency on "go/token"
FAIL
FAIL	cmd/compile/internal/gc	40.383s
```

I noticed this on the current `release-branch.go1.13` branch, but it might be a problem on `master` too.

/cc @andybons

I don't know how serious of a problem this is, we should find who has more context on this test. It's new to 1.13.

**Comment by dmitshur** [comment]
Test was added in [CL 172997]() to fix issue #31563. /cc @josharian @bradfitz

It seems like a very minor optimization, so not a release blocker.

Additionally, this _might_ be a false positive. The test runs `exec.Command("go", ...)`, which I suspect picked up Go 1.12.9 in my PATH instead of the `release-branch.go1.13` version I was running tests on.

**Comment by dmitshur** [comment]
It was a false positive. The test needs to be improved.

I can reproduce on `master` branch:

```
$ ./make.bash
[...]
$ ../bin/go version
go version devel +3c56143442 Tue Sep 3 17:10:37 2019 +0000 darwin/amd64
$ ../bin/go test cmd/compile/internal/gc
--- FAIL: TestDeps (0.16s)
    dep_test.go:23: undesired dependency on "go/build"
    dep_test.go:23: undesired dependency on "go/token"
FAIL
FAIL	cmd/compile/internal/gc	9.685s
FAIL
$ go version
go version go1.12.9 darwin/amd64
$ PATH="$(pwd)/../bin:$PATH" go version
go version devel +3c56143442 Tue Sep 3 17:10:37 2019 +0000 darwin/amd64
$ PATH="$(pwd)/../bin:$PATH" go test cmd/compile/internal/gc
ok  	cmd/compile/internal/gc	7.000s
```
